File: Lab3/listLab/listLab/Source.cpp
#include <iostream>
#include <fstream>
#include <chrono>
#include <list>
#include <iterator>
#include <math.h>

using namespace std;
using namespace chrono;

int main()
{
    high_resolution_clock::time_point start_time, end_time;

    ifstream in; 
    in.open("numbers.txt");

    list<int> numlist;
    list<int>::iterator curelem;

    int num;

    start_time = high_resolution_clock::now();

    do
    {
        in >> num;
        numlist.push_back(num);
    } while (!in.eof());

    curelem = numlist.begin();
    int a = *curelem;
    curelem++;
    int b = *curelem;
    curelem++;
    int ab, k = 0;
    for (curelem; curelem != numlist.end(); curelem++)
    {
        ab = *curelem;
        if (((a + b) % 3 == 0) && ((a + b) % 6 != 0) && ((a * b) % 10 == 8))
            k++;
        a = b;
        b = ab;
    }

    end_time = high_resolution_clock::now();

    cout << k << endl;
    cout << duration_cast<milliseconds>(end_time - start_time).count() << " ms";
    in.close();
    return 0;
}

File: Lab3/Source.py
from datetime import datetime

f = open("numbers.txt")

start_time = datetime.now()

numlist = [int(x) for x in f.readlines()]

k = 0
for i in range(len(numlist)-2):
    a = numlist[i]
    b = numlist[i+1]
    if ((a + b) % 3 == 0) and ((a + b) % 6 != 0) and ((a * b) % 10 == 8):
        k += 1

end_time = datetime.now()

print(k)
print((end_time - start_time).microseconds // 1000)
f.close()
